impl.go

```diff
@@ -305,6 +320,32 @@ func validReceiver(recv string) bool {
 	return err == nil
 }
 
+// commentsBefore returns true if comments precede the field. It returns false
+// otherwise.
+func commentsBefore(field *ast.Field, commentGroups []*ast.CommentGroup) bool {
+	if len(commentGroups) > 0 {
+		return commentGroups[0].Pos() < field.Pos()
+	}
+	return false
+}
+
+// flattenCommentMap flattens the comment map to a string.
+func flattenCommentMap(m ast.CommentMap) string {
+	var result strings.Builder
+	for _, cmmtgroups := range m {
```

Comment:
Nit: s/cmmtgroups/cgs/ and s/cmmtgroup/cg/ and s/cmt/c/.

Reply:
Is m guaranteed to have only one element at this point? If so (as I suspect), please add a comment to that effect. If not, explain why this will not cause indeterministic output?

Reply:
Done.